core/src/Cabana_Experimental_NeighborList.hpp

```diff
@@ -165,17 +164,39 @@ struct NeighborDiscriminatorCallback2D_FirstPass
     }
 };
 
+// Preallocate and attempt fill in the first pass
+template <typename Counts, typename Neighbors, typename Tag>
+struct NeighborDiscriminatorCallback2D_FirstPass_BufferOptimization
+{
+    Counts counts;
+    Neighbors neighbors;
+    using tag = ArborX::Details::InlineCallbackTag;
+    template <typename Predicate>
+    KOKKOS_FUNCTION void operator()( Predicate const &predicate,
+                                     int primitive_index ) const
+    {
+        int const predicate_index = getData( predicate );
+        if ( CollisionFilter<Tag>::keep( predicate_index, primitive_index ) )
+        {
+            if ( counts( predicate_index ) < (int)neighbors.extent( 1 ) )
```

Comment:
From the performance perspective, you don't want to call extent all the time here. Why not store it as a member?

Reply:
I don't think it matters although I have no more evidence than you do.  I am not changing.